File: Ruby/problem004.rb
#! /usr/bin/env ruby
=begin
Problem 4 - Project Euler
http://projecteuler.net/index.php?section=problems&id=4
=end
class Integer
    def palindromic?
        self == self.to_s.reverse.to_i
    end
end

puts (100..999).to_a
        .combination(2)
        .map {|a, b| a * b}
        .select {|x| x.palindromic? }
        .max

File: Nim/problem004.nim
# Problem 4 - Project Euler
# http://projecteuler.net/index.php?section=problems&id=4
from strutils import parseInt
import unicode

proc palindrome(p: int): bool =
    return p == reversed($p).parseInt

var ans = 0
for i in 100..999:
    for j  in i..999:
        echo i
        let p = i * j
        if palindrome(p):
            ans = max(ans, p)

echo ans
